src-packed/validator.js

```diff
@@ -62,12 +63,31 @@ export async function getMatches(text, matches) {
         console.log(result.error);
         return;
     }
-    result.sentences.forEach(sentence => {
+
+    for (var i = 0; i < result.sentences.length; i++) {
+        var sentence = result.sentences[i];
         if (sentence.sentiment !== "negative")
-            return;
+            continue;
         console.log(`Index: ${sentence.offset}, Negative sentiment: ${sentence.confidenceScores.negative}`)
         if (sentence.confidenceScores.negative <= 0.75)
-            return;
+            continue;
+
+        // Remove commonly used phrases with a negative sentiment
+        var regex = new RegExp(ignoreList.ignorablePhrases.join("|"), "gi");
+        var trimmedText = sentence.text.replace(regex, "");
+        // Analyze sentiment again if a commonly used phrase with a negative sentiment was removed
+        if (trimmedText !== sentence.text) {
+            const trimmedResult = await textAnalytics.analyzeSentiment(trimmedText);
+            if (trimmedResult.error !== undefined) {
+                console.log(trimmedResult.error);
+                continue;
+            }
+            // If trimmed sentence is no longer considered negative, return
+            if (trimmedResult.sentences[0].sentiment !== "negative")
+                continue;
```

Comment:
If you want to keep using `forEach()`, I think we can use `return;` instead?

I don't know which one is more JavaScript-y, I'm just guessing really.

Reply:
From the brief reading I have done it seems that `forEach` and async/await aren't very compatible, but if I change this back to use a single Text Analytics API call I think we can continue to use `forEach`.